finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Parameter.scala

```diff
@@ -0,0 +1,41 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.BufferWriter
+import language.implicitConversions
+
+/**
+ * A value of type `A` can implicitly convert to a `Parameter` if an evidence `CanBeParameter[A]` is
+ * available in scope. This type is not be instantiated in any other manner.
+ */
+sealed trait Parameter {
+  type A
+  def value: A
+  def evidence: CanBeParameter[A]
+
+  final def writeTo(writer: BufferWriter): Unit = {
+    evidence.write(writer, value)
+  }
+
+  final def size: Int = evidence.sizeOf(value)
+  final def typeCode: Short = evidence.typeCode(value)
+}
+
+object Parameter {
+  implicit def wrap[A0](value0: A0)(implicit evidence0: CanBeParameter[A0]): Parameter = {
```

Comment:
I see. With this, though, the client is burdened with code like following:

```scala
client.prepare("...").apply(3.asParam, "jd".asParam)
```

I think it adds no value and hampers readability.